src/libraries/System.Security.Cryptography/tests/X509Certificates/CertLoaderTests.cs

```diff
@@ -0,0 +1,139 @@
+        {
+            protected override X509Certificate2 LoadCertificate(byte[] data)
+            {
+                string path = Path.GetTempFileName();
+
+                try
+                {
+                    File.WriteAllBytes(path, data);
+                    return X509CertificateLoader.LoadCertificateFromFile(path);
+                }
+                finally
+                {
+                    File.Delete(path);
+                }
+            }
+        }
+
+        public class LegacyLoaderFromArray : CommonTests
+        {
+            protected override X509Certificate2 LoadCertificate(byte[] data) =>
+                new X509Certificate2(data);
+        }
+
+        [ActiveIssue("NestedCertificates tests fail", TestPlatforms.Linux)]
+        public class LegacyLoaderFromFile : CommonTests
+        {
+            protected override X509Certificate2 LoadCertificate(byte[] data)
+            {
+                string path = Path.GetTempFileName();
+
+                try
+                {
+                    File.WriteAllBytes(path, data);
+                    return new X509Certificate2(path);
+                }
+                finally
+                {
+                    File.Delete(path);
+                }
+            }
+        }
+
+        public abstract class CommonTests
+        {
+            protected abstract X509Certificate2 LoadCertificate(byte[] data);
+
+            [Fact]
+            public void LoadWrappingCertificate_DER()
+            {
+                using (X509Certificate2 cert = LoadCertificate(TestData.NestedCertificates))
+                {
+                    AssertExtensions.SequenceEqual(TestData.NestedCertificates, cert.RawDataMemory.Span);
+                }
+            }
+
+            [Fact]
+            public void LoadWrappingCertificate_PEM()
+            {
+                byte[] data = System.Text.Encoding.ASCII.GetBytes(
+                    PemEncoding.Write("CERTIFICATE", TestData.NestedCertificates));
+
+                using (X509Certificate2 cert = LoadCertificate(data))
+                {
+                    AssertExtensions.SequenceEqual(TestData.NestedCertificates, cert.RawDataMemory.Span);
+                }
+            }
+
+            [Fact]
+            public void LoadWrappingCertificate_DER_Trailing()
+            {
+                byte[] source = TestData.NestedCertificates;
+                byte[] data = new byte[source.Length + 5];
+                Array.Copy(source, 0, data, 0, source.Length);
+                RandomNumberGenerator.Fill(data.AsSpan(source.Length));
+
+                using (X509Certificate2 cert = LoadCertificate(data))
+                {
+                    AssertExtensions.SequenceEqual(TestData.NestedCertificates, cert.RawDataMemory.Span);
+                }
+            }
+
+            [Fact]
+            [ActiveIssue("Fails as NewFile, NewSpan, NewArray, LegacyArray", TestPlatforms.Linux)]
```

Comment:
Are these `ActiveIssue` tests things that need to be addressed? If they will be addressed after this pull request, can we write an issue describing the problem and add a link back here?

Same feedback for the rest of the file.